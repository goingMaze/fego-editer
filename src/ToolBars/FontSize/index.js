import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { RichUtils } from 'draft-js'
import { ToolBarBtn, Select, Func } from 'utils'
let { toggleInlineStyle } = Func;

export default class FontSize extends Component {
	static propTypes = {
		editorState: PropTypes.object.isRequired,
		onChange: PropTypes.func.isRequired,
		config: PropTypes.array.isRequired,
		modalManage: PropTypes.object.isRequired
	}
	constructor(props) {
		super(props)
		this.state = {
			expanded: false
		}
	}
	componentWillMount() {
		let { modalManage } = this.props;
		modalManage.addCallback(this.changeExpand)
	}
	componentWillUnmount() {
		let { modalManage } = this.props;
		modalManage.removeCallback(this.changeExpand)
	}
	changeExpand = () => {
		this.setState({
			expanded: this.expanded
		})
		this.expanded = false
	}
	onExpand = () => {
		this.expanded = !this.state.expanded
	}
	onToggle = inlineStyle => {
		let { editorState, onChange, config } = this.props;
		config = config.map(item => `fontSize${item}`)
		let newEditorState = toggleInlineStyle(editorState, config, inlineStyle)
		onChange(newEditorState)
	}
	render() {
		let { editorState, config, modalManage } = this.props;
		let { expanded } = this.state;
		let currentStyle = editorState.getCurrentInlineStyle();
		let activeItem = 0;
		let options = config.map(item => ({
			style: `fontSize${item}`,
			label: item
		}))
		options.find((item, idx) => {
			if(currentStyle.has(item.style)) {
				activeItem = idx
				return true
			}
			return false
		})
		return (
			<div className="FegoEditor-toolbars" title='字号' >
				<Select {...{ activeItem, expanded, options }} onToggle={this.onToggle} changeExpand={this.onExpand} />
			</div>
		)
	}
}