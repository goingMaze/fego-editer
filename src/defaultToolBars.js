const defaultToolBars = {
	options: ['BlockType', 'Color', 'FontFamily', 'FontSize', 'Image', 'Inline', 'Link', 'List', 'TextAlign', 'Video', 'History'],
	BgColor: [],
	BlockType: ['正文', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'],
	Color: ['rgb(97,189,109)', 'rgb(26,188,156)', 'rgb(84,172,210)', 'rgb(44,130,201)',
		'rgb(147,101,184)', 'rgb(71,85,119)', 'rgb(204,204,204)', 'rgb(65,168,95)', 'rgb(0,168,133)',
		'rgb(61,142,185)', 'rgb(41,105,176)', 'rgb(85,57,130)', 'rgb(40,50,78)', 'rgb(0,0,0)',
		'rgb(247,218,100)', 'rgb(251,160,38)', 'rgb(235,107,86)', 'rgb(226,80,65)', 'rgb(163,143,132)',
		'rgb(239,239,239)', 'rgb(255,255,255)', 'rgb(250,197,28)', 'rgb(243,121,52)', 'rgb(184,49,47)',
		'rgb(209,72,65)', 'rgb(124,112,107)', 'rgb(209,213,216)'],
	FontFamily: [{
		name: 'SimHei', family: '"SimHei", "黑体"'
	}, {
		name: 'Yahei', family: '"Microsoft Yahei", "微软雅黑"'
	}, {
		name: 'KaiTi', family: '"KaiTi", "楷体"'
	}, {
		name: 'FangSong', family: '"FangSong", "仿宋"'
	}],
	FontSize: [16, 20, 30, 40, 50, 70, 100, 200],
	History: ['undo', 'redo'],
	Image: {
		uploadUrl: '',
		uploadCallback: null
	},
	Inline: ['bold', 'italic', 'underline'],
	Link: {},
	List: ['ul', 'ol', 'indent', 'outdent'],
	TextAlign: ['left', 'center', 'right', 'justify'],
	Video: {},
	Remove: {}
};

export default defaultToolBars
