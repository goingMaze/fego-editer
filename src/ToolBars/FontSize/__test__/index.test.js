import React from 'react'
import { mount, render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { EditorState, ContentState, convertFromHTML } from 'draft-js'
import FontSize from '..'
import defaultToolBars from '../../../default'
import ModalManage from '../../../utils/ModalManage'

describe('FontSize', () => {
	const contentState = ContentState.createFromBlockArray(convertFromHTML('<div>test</div>'))
	const editorState = EditorState.createWithContent(contentState)
	const wrapper = mount(
		<FontSize editorState={editorState} onChange={() => { }}
			config={defaultToolBars.FontSize} modalManage={new ModalManage()}
		/>
	);
	it('渲染时开头应该有一个div', () => {
		expect(wrapper.html().startsWith('<div')).toEqual(true)
	})
	it('应该有一个Select子组件', () => {
		expect(wrapper.find('Select').length).toBe(1)
	})
	it('config应该为默认值', () => {
		expect(wrapper.prop('config')).toEqual(defaultToolBars.FontSize)
	})
	it('在expanded为true时应该有8个li元素', () => {
		expect(wrapper.find('li').length).toBe(0)
		wrapper.setState({
			expanded: true
		})
		expect(wrapper.find('li').length).toBe(8)
	})
	it('快照', () => {
		const rendered = render(
			<FontSize editorState={editorState} onChange={() => { }}
				config={defaultToolBars.FontSize} modalManage={new ModalManage()}
			/>
		)
		expect(toJSON(rendered)).toMatchSnapshot()
	})
})