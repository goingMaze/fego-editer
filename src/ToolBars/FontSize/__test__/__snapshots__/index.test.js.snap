// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontSize 快照 1`] = `
<div
  class="FegoEditor-toolbars"
  title="字号"
>
  <div
    class="FegoEditor-select"
  >
    <div
      class="FegoEditor-selectLabel"
    >
      <div>
        16
      </div>
      <svg
        aria-hidden="true"
        class="icon"
        style="vertical-align:top;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M494 638l256.5-252H242z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`FontSize 快照 1`] = `
<div
  class="FegoEditor-toolbars"
  title="字号"
>
  <div
    class="FegoEditor-select"
  >
    <div
      class="FegoEditor-selectLabel"
    >
      <div>
        16
      </div>
      <svg
        aria-hidden="true"
        class="icon"
        style="vertical-align:top;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M494 638l256.5-252H242z"
        />
      </svg>
    </div>
  </div>
</div>
`;
